server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -579,4 +588,15 @@ public boolean isConsumeTimeoutEnabled() {
     public long getConsumeTimeout() {
         return consumeTimeout;
     }
+
+    /**
+     * It cleans up resources after receiving the SIGTERM signal. E.g. closing Kafka connections
+     */
+    private void registerShutDownHook() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            this.stop();
+            this.connNorm.wakeup();
+            logger.info("Cleaned up resources before shutting down the JVM");
```

Comment:
This message is incorrect... the WakeupException only breaks long-running polls, which allows the plugin to exit timely. Maybe just remove the message.